wandb/data_types.py

```diff
@@ -490,20 +490,26 @@ def _get_updated_result_type(self, row):
             )
         return result_type
 
-    def _to_table_json(self, max_rows=None, warn=True):
+    def _to_table_json(self, max_rows=None, enforce_max_row_limit=True):
         # separate this method for easier testing
         if max_rows is None:
             max_rows = Table.MAX_ROWS
-        if len(self.data) > max_rows and warn:
-            logging.warning("Truncating wandb.Table object to %i rows." % max_rows)
+        n_rows = len(self.data)
+        if n_rows > max_rows and enforce_max_row_limit:
+            raise ValueError(
+                f"Table row limit exceeded: table has {n_rows} rows, limit is {max_rows}. "
+                f"To increase the maximum number of allowed rows in a wandb.Table, override "
+                f"the limit with `wandb.Table.MAX_ARTIFACT_ROWS = X` and try again. Note: "
+                f"this may cause slower queries in the W&B UI."
+            )
         return {"columns": self.columns, "data": self.data[:max_rows]}
 
     def bind_to_run(self, *args, **kwargs):
         # We set `warn=False` since Tables will now always be logged to both
```

Comment:
Maybe remove this documentation line since no `warn` kwarg exists anymore?

Reply:
nice - thanks. done